draft-ietf-sidrops-publication-server-bcp-01.md

```diff
@@ -139,6 +139,70 @@ of the root cause for disruption in the Publication Server that effectively is
 part of their infrastructure, and helps publishers avoid - if possible -
 changes in published RPKI objects that are needed during these windows.
 
+## Availability and Data Loss
+
+The Publication Server MUST ensure high availability of the [@!RFC8181]
+publication protocol. If this protocol is not available then this will prevent
+publishers from updating their ROAs and renewing their manifests and CRLs.
+
+Publication Servers MUST aim to minimise data loss in case of severe server
+outages. If a server restore is needed and a content regression has occured due
+to data loss then the server MUST perform an RRDP session reset.
+
+Publishing CAs typically only check in with their Publication Server when they
+have changes that need to be published. As a result they may not be aware if the
+server performed a restore and their content regressed to an earlier state. This
+could result in two problems:
+
+ - The ROAs may not reflect what the publisher intended
+ - The publisher may not renew their manifest or CRL in time, because they
+   assume that their current manifest and CRL have not yet expired or become
+   stale
+
+Therefore, the Publication Server SHOULD notify publishing CAs about this issue
+if it occurs, so that a full manually triggered resynchronisation can then be
+initiated by CAs.
+
+## Publisher Repository Synchronisation
+
+### Full Synchronisation
+
+It is RECOMMENDED that publishing CAs always perform a list query as described
+in section 2.3 of [@!RFC8181] before sending all their changes using multiple
+PDUs as described in section 2.2 of [@!RFC8181]. This way any desynchronisation
+issue can be resolved at least as soon as the publisher is aware of updates that
+it needs to publish.
+
+In addition to this the publishing CA MAY perform regular planned
+synchronisation events where it issues an [@!RFC8181] list query even if it has
+no new content to publish. Because this interaction requires that the
+Publication Server signs an [@!RFC8181] list reply, this operation can be costly
+for Publication Servers that serve a large number of publishers. Therefore,
```

Comment:
Why is this costly?  Is it possible that this is a particular problem with Krill because it's generating a new EE certificate for each transaction, as it does for the RFC 6492 service?

Reply:
I think it's mostly okay, but in case of 1500+ CAs (which happens) it does get rather chatty..

We spoke about this in the past and then indeed there was a problem in Krill, but that was something else - there was an inefficiency in the implementation that meant it spent lots of time on deserializing state - that has been fixed.

Nonetheless, 1500 signed responses every minute could be significant and because the publication protocol does not (yet) support rate limiting there would not be a clean way to tell clients to back off if needed.

So, therefore I thought... well getting the notification file just hits the CDN, so there is no problem in hammering that from delegated CAs, which will likely be fewer in number than RPs anyway.

Reply:
To be clear: I am open to suggestions :) How about:

```
Because this interaction requires that the Publication Server signs an [@!RFC8181] list reply, this operation can be costly for Publication Servers that serve a large number of publishers.
```

New:
```
For Publication Server that serve a large number (1000s) of publishers this operation could become costly, and unfortunately the [@!RFC8181] protocol has no clean support for rate limiting.
```

Would that work?

Reply:
Yep, I think mentioning the rate limiting consideration is a good idea.  The suggested text above sounds fine to me.

Reply:
[Fixed in this commit](https://github.com/timbru/draft-sidrops-publication-server-bcp/pull/30/commits/28e14f33ffe2ac568858849361bc902d35dacf77)